Terminal.Gui/Views/Button.cs

```diff
@@ -145,7 +145,13 @@ internal void Update ()
 				base.Text = ustring.Make (_leftBracket) + " " + text + " " + ustring.Make (_rightBracket);
 
 			int w = base.Text.RuneCount - (base.Text.Contains (HotKeySpecifier) ? 1 : 0);
-			Width = w;
+			try {
+				Width = w;
+#pragma warning disable RCS1075 // Avoid empty catch clause that catches System.Exception.
```

Comment:
I don't think DimPos should be throwing an exception in this case. There should be no issue with assigning to `Width` here. I think the right fix is to back out the code that was added to `Width` in Commit 5efa8d53 to throw this exception, or modify it to not throw in this case.

Reply:
This is debatable. Imagine that you have a button anchored in a position that depends on other views but you need to change its caption from time to time. This will stop the button from being anchored and force the user to reconfigure the anchorage again. If it is a custom control the user will have to solve this but with native controls that are part of the `Terminal.Gui` I do not know if this should be avoided so as not to cause future annoyances for the user to wonder why my button is no longer anchored for just changing the caption?

Reply:
I see your point now. I think.

I need to ponder this some more.